### PR TITLE
tests: fix issue when checking the udev tag on test security-device-cgroups

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes:
. When the rule is added, then teh TAGS and CURRENT_TAGS are updated
. Remove the snap and check the udev CURRENT_TAGS is remove
. After rebooting the system the udev TAGS is removed as well

To reproduce the error run:
google:ubuntu-21.04-64:tests/main/security-device-cgroups:uinput
google:ubuntu-21.04-64:tests/main/security-device-cgroups:kmsg

The error:
+ echo 'And other devices are not shown as assigned to the snap'
And other devices are not shown as assigned to the snap
+ not MATCH 'E: TAGS=.*snap_test-snapd-sh_sh'
+ udevadm info /sys/devices/virtual/misc/uinput

Full log: https://paste.ubuntu.com/p/CCtdrNTXPC/

### CODE REVIEW

File: tests/main/security-device-cgroups/task.yaml [hunk 1]
```diff
@@ -77,50 +80,62 @@ execute: |
...
-    # TODO: check device unassociated after removing the udev file and rebooting
+        echo "Once the snap is removed, the current tags are automatically removed"
+        snap remove test-snapd-sh
+        udevadm info "$UDEVADM_PATH" | not MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
+
+        rm -f /etc/udev/rules.d/70-snap.test-snapd-sh.rules
+        REBOOT
+    fi
+
+    udevadm info "$UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
```
Comment (anonymouse64) [inline tests/main/security-device-cgroups/task.yaml:141]: can we use `NOMATCH` here?

**Review by anonymouse64 — COMMENTED** [review]
lgtm if the tests are green, one possible small improvement

File: tests/main/security-device-cgroups/task.yaml [hunk 2]
```diff
@@ -77,50 +80,62 @@ execute: |
...
-    "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        echo "Given a snap is installed"
+        "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
 
-    echo "Then the device is not assigned to that snap"
-    udevadm info "$UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
+        echo "Then the device is not assigned to that snap"
+        udevadm info "$UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
+        udevadm info "$UDEVADM_PATH" | not MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
```
Comment (bboozzoo) [inline tests/main/security-device-cgroups/task.yaml:89]: This will only work with systemd 247+, released in Nov 2020, so Ubuntu 21.04 and later

Comment (sergiocazzolato) [reply tests/main/security-device-cgroups/task.yaml]: Nice, thanks for that info, I already added a check for the system version

File: tests/main/security-device-cgroups/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/security-device-cgroups/task.yaml]: sorry, because copy&paste. Updated all the "not MATCH" in that test.
Thanks for catching this

**Review by anonymouse64 — APPROVED** [review]
lgtm, thanks for the changes

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**